Source/Core/Core/AchievementManager.cpp

```diff
@@ -270,6 +273,82 @@ void AchievementManager::ActivateDeactivateRichPresence()
       nullptr, 0);
 }
 
+void AchievementManager::FetchBadges()
+{
+  if (!m_is_runtime_initialized || !IsLoggedIn() || !Config::Get(Config::RA_BADGES_ENABLED))
+    return;
+  if (!m_player_badge.loaded)
+  {
+    m_image_queue.EmplaceItem([this] {
+      {
+        std::lock_guard lg{m_lock};
+        std::string username = Config::Get(Config::RA_USERNAME);
+        rc_api_fetch_image_request_t icon_request = {.image_name = username.c_str(),
+                                                     .image_type = RC_IMAGE_TYPE_USER};
+        if (RequestImage(icon_request, &m_player_badge) == ResponseType::SUCCESS)
+          INFO_LOG_FMT(ACHIEVEMENTS, "Successfully downloaded player badge.");
+        else
+          WARN_LOG_FMT(ACHIEVEMENTS, "Failed to download player badge.");
```

Comment:
For this warning to be useful, I'd imagine you would want to put details that a developer might want to see to diagnose errors.  Maybe image name?